src/EditorFeatures/Core/GoToBase/AbstractGoToBaseService.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.FindUsages;
+
+namespace Microsoft.CodeAnalysis.Editor.GoToBase
+{
+    internal abstract partial class AbstractGoToBaseService : IGoToBaseService
+    {
+        public async Task FindBasesAsync(Document document, int position, IFindUsagesContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var tuple = await FindBaseHelpers.FindBasesAsync(document, position, cancellationToken).ConfigureAwait(false);
+            if (tuple == null)
+            {
+                await context.ReportMessageAsync(
+                    EditorFeaturesResources.Cannot_navigate_to_the_symbol_under_the_caret).ConfigureAwait(false);
+                return;
+            }
+
+            var message = tuple.Value.message;
+
+            if (message != null)
+            {
+                await context.ReportMessageAsync(message).ConfigureAwait(false);
+                return;
+            }
+
+            await context.SetSearchTitleAsync(
+                string.Format(EditorFeaturesResources._0_implementations,
+                FindUsagesHelpers.GetDisplayName(tuple.Value.symbol))).ConfigureAwait(false);
+
+            var solution = document.Project.Solution;
+
+            foreach (var implementation in tuple.Value.implementations)
```

Comment:
if there's only one impl, should we just be navigating directly to it?

Reply:
Shouldn't this be implemented on the command handler layer ([https://github.com/dotnet/roslyn/blob/bd1e0c6cb23e552d390c1b3c14cdb6ec8434883e/src/EditorFeatures/Core/FindReferences/FindReferencesCommandHandler.cs#L63](https://github.com/dotnet/roslyn/blob/bd1e0c6cb23e552d390c1b3c14cdb6ec8434883e/src/EditorFeatures/Core/FindReferences/FindReferencesCommandHandler.cs#L63))?

---
In reply to: [283972057](https://github.com/dotnet/roslyn/pull/35704#discussion_r283972057) [](ancestors = 283972057)